### ISSUE TITLE
(right) Click on object input to select a sprite?

### ISSUE THREAD

**Issue by liam4** [jmoenig/Snap--Build-Your-Own-Blocks#1200]
Following up on what cycomachead [said](https://github.com/jmoenig/Snap--Build-Your-Own-Blocks/issues/1199#issuecomment-216336298):

> > Let's not change ASK and TELL, because you can still just type sprite names into them, which is actually really nice sometimes when you already know who you want to address
>
> Yeah, I guess that's true. I always found the sprite icon really intuitive for what goes in the slot. (Though I guess the word "Sprite" is too, it's just not as cute.)

---

Take a look at this block:

![tell-obj-to](https://cloud.githubusercontent.com/assets/9948030/14965420/bb39f04e-1083-11e6-8b50-8fe0be2c6394.png)

It's a modified version of TELL, to use an object input instead of a text input. As Jens stated, it's handy to be able to type in the name of a sprite:

![tell-alonzo-to-say-hi](https://cloud.githubusercontent.com/assets/9948030/14965597/a5aabfc8-1084-11e6-9a08-0bb7187109d6.png)

..instead of having to do something less, you know, nice:

![tell-sprite-whos-name-is-alonzo-to-say-hi](https://cloud.githubusercontent.com/assets/9948030/14965556/71f7a38a-1084-11e6-833b-a475b71e6e53.png)

I wonder if it would be better if you could click (or perhaps right-click) to show a context menu that allows you to select a sprite?

It might look something like this, after you've selected a sprite:

![tell-alonzo-to](https://cloud.githubusercontent.com/assets/9948030/14965625/ca4dc9b0-1084-11e6-922b-554ee202e3dd.png)

It might also be nice to have a picture of Alonzo beside the sprite name, and perhaps a little help balloon that appears when you hover over the input that says "(right) click to select a sprite", but I'm not sure about any of those details myself. You decide if you want.

Any comments?

**Comment by jmoenig** [comment]
this is sorta on the list of things that are coming in the not too far away future: custom drop-down menus that you can define with Snap reporters (instead of using fixed key-value pairs like now). Then you'll be able to build your own sprite-names drop-down input slot.

**Comment by cycomachead** [comment]
> Then you'll be able to build your own sprite-names drop-down input slot.

Or, what we should really do sometime is allow custom blocks to access the primitive dropdown options. There's no reason a sprite block couldn't access `%spr` menu, or list blocks couldn't make use of the same nice menus as the primitives.

**Comment by brianharvey** [comment]
I agree that %spr would be nice, but those lists should be available programmatically (get other sprites, etc.) too.

**Comment by cycomachead** [comment]
> I agree that %spr would be nice, but those lists should be available programmatically (get other sprites, etc.) too.

I'd support that! If they were programmatically available, then you could use them in a world where a reporter can generate menus. 👍  (But Jens doesn't like my pull request getting us towards this world... :P #1109)
